pkg/kube/cluster-init.sh

```diff
@@ -204,6 +205,15 @@ config_cluster_roles() {
 }
 
 check_start_k3s() {
+  # If cluster is in transition, wait until transition is complete
+  if [ "$WAIT_FOR_CLUSTER_TRANSITION" = "true" ]; then
+        logmsg "Cluster transition in progress, waiting before starting k3s"
+        while [ "$WAIT_FOR_CLUSTER_TRANSITION" = "true" ]; do
+                sleep 2
```

Comment:
FWIW, you can use named pipes to implement this kind of notification, instead of sleep over variables... for example:

```sh
#!/bin/sh

NP="/tmp/fifo$$"
mkfifo "$NP"

funcA() {
	echo "Waiting for funcB()..."
	read lock < "$NP"
	echo "funcA: Done"
}

funcB() {
	echo "Sleeping before notify funcA..."
	sleep 5
	echo "WAKEUP!" > "$NP"
	echo  "funcB: Done"
}

funcB &
funcA

rm "$NP"
```

Reply:
good suggestion, will do.